ruby/red-arrow/test/test-array-builder.rb

```diff
@@ -146,6 +146,16 @@ def assert_build(builder_class, raw_array)
                        ["Apache Arrow"],
                      ])
       end
+
+      test("list<uint>s") do
+        array = Arrow::ArrayBuilder.build([[0, 1, 2], [3, 4]])
+        assert_equal("list<item: uint8>", array.value_data_type.to_s)
```

Comment:
Could you also check values like https://github.com/apache/arrow/blob/1fac1317dc6aaeaf09e35d2363160d0b7a80442b/ruby/red-arrow/test/test-array-builder.rb#L97-L108 ?